docs/source/concepts.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Concepts
+---
+<h2 id="concepts">Link Concepts</h2>
+
+Apollo Link is designed to be a powerful way to compose actions around data handling with GraphQL. Each link represents a subset of functionality that can be composed with other links to create complex control flows of data. At a basic level, a link is a function that takes an operation and returns an observable. Described with types, it looks like this:
+
+```js
+type Context = { [key: string]: any };
+
+interface Operation {
+  query: DocumentNode;
+  variables: { [key: string]: any };
+  operationName: string;
+  extensions?: { [key: string]: any };
+  getContext(): Context;
+  setContext(newContext: Context | (prevContext: Context) => Context): void;

```

Comment:
I think we could get rid of the function argument API.

Reply:
Originally I had but when I was writing the tests it was pretty helpful. It was easier then getting the context then setting it on the next line.

Reply:
Dang. It just feels a bit weirdly duplicative.